api/desecapi/migrations/0041_rename_user_token_owner_token_user_override_and_more.py

```diff
@@ -0,0 +1,98 @@
+# Generated by Django 5.1.3 on 2024-11-25 19:53
+
+import django.db.models.deletion
+import django.db.models.functions.comparison
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("desecapi", "0040_token_auto_policy_token_token_auto_policy_and_more"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="ALTER TABLE desecapi_tokendomainpolicy DROP CONSTRAINT desecapi_tokendomainpolicy_token_id_token_user_id_fkey RESTRICT;",
+            reverse_sql="",
+        ),
+        migrations.RemoveConstraint(
+            model_name="token",
+            name="unique_id_user",
+        ),
+        migrations.AlterField(
+            model_name="token",
+            name="user",
+            field=models.UUIDField(),
+        ),
+        migrations.RenameField(
+            model_name="token",
+            old_name="user",
+            new_name="owner",
+        ),
+        migrations.AlterField(
+            model_name="token",
+            name="owner",
+            field=models.ForeignKey(
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="+",
+                to=settings.AUTH_USER_MODEL,
+            ),
+        ),
+        migrations.AddField(
+            model_name="token",
+            name="user_override",
+            field=models.ForeignKey(
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="+",
+                to=settings.AUTH_USER_MODEL,
+            ),
+        ),
+        migrations.AddConstraint(
+            model_name="token",
+            constraint=models.CheckConstraint(
+                condition=models.Q(
+                    ("user_override__isnull", True),
+                    models.Q(("mfa__isnull", True), ("perm_manage_tokens", False)),
+                    _connector="OR",
+                ),
+                name="user_override_conditions",
+            ),
+        ),
+        migrations.AddField(
+            model_name="token",
+            name="user_id",
+            field=models.GeneratedField(
+                db_index=True,
+                db_persist=True,
+                expression=django.db.models.functions.comparison.Coalesce(
+                    "user_override", "owner"
+                ),
+                output_field=models.UUIDField(),
+            ),
+        ),
+        migrations.AddConstraint(
+            model_name="token",
+            constraint=models.UniqueConstraint(
+                fields=("id", "user_id"), name="unique_id_user"
+            ),
+        ),
+        migrations.RunSQL(
+            sql="ALTER TABLE desecapi_tokendomainpolicy ADD FOREIGN KEY ( token_id, token_user_id ) REFERENCES desecapi_token ( id, user_id );",
```

Comment:
how does the name match up with the dropped constraint above?

Reply:
The name is generated automatically, and I've verified that it does match up.